naga/src/back/msl/writer.rs

```diff
@@ -3288,7 +3288,9 @@ impl<W: Write> Writer<W> {
         {
             let mut indices = vec![];
             for (handle, var) in module.global_variables.iter() {
-                if needs_array_length(var.ty, &module.types) {
+                if pipeline_options.experimental_vertex_pulling_transform
```

Comment:
I don't think this is right, we have to instead generate one field for each vertex buffer in `pipeline_options.vertex_buffer_mappings`.

Reply:
Is that because the buffers are not necessarily global variables? I guess I need to understand better how this structure is supplied to the entry point at the point of calling.

Reply:
They are represented as global variables at the IR level but `module.global_variables` won't contain the vertex buffers we bound for vertex pulling since we do that after we generate the IR (i.e. `global_variables` gets populated earlier by the WGSL front-end).

Reply:
Updated to get the buffer indices from `pipeline_options.vertex_buffer_mappings`. Still need to figure out how to supply this data when the entry point is invoked.